### PR TITLE
created searchbar component and added to navbar

### PR DESCRIPTION
**Pull request by Star3Lord**

### CODE REVIEW

File: front-end/src/pages/profile.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ export default function Profile({ picture }) {
 
 export async function getServerSideProps() {
   const picture = await axios
-    .get(process.env.PICTURE_API, {
+    .get("https://picsum.photos/200/300", {
```
Comment (Verse1) [inline front-end/src/pages/profile.js:109]: Don't paste urls directly even if it's a mock API pls

File: front-end/src/styles/Navigation.css [hunk 2]
```diff
@@ -1,3 +1,7 @@
-  .link {
-    @apply inline-block py-2 px-4 text-white no-underline hover:text-violet-400;
-  }
\ No newline at end of file
+.link {
+  @apply inline-block py-2 px-4 text-white no-underline hover:text-violet-400;
+}
+
+.search-bar {
```
Comment (Verse1) [inline front-end/src/styles/Navigation.css:5]: Could you maybe make the search bar smaller

File: front-end/src/components/SearchBar.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { Fragment } from 'react';
+import { FaSearch } from 'react-icons/fa';
+
+const Search = () => {
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log({ search: e.target.search.value });
```
Comment (Verse1) [inline front-end/src/components/SearchBar.jsx:7]: What is this really accomplishing for the user?

**Review by Verse1 — CHANGES_REQUESTED** [review]

File: front-end/src/components/SearchBar.jsx (hunk 3 above)
Comment (Star3Lord) [reply front-end/src/components/SearchBar.jsx:7]: search classes once the backend is setup

File: front-end/src/styles/Navigation.css (hunk 2 above)
Comment (Star3Lord) [reply front-end/src/styles/Navigation.css:5]: the length?

File: front-end/src/pages/profile.js (hunk 1 above)
Comment (Star3Lord) [reply front-end/src/pages/profile.js:109]: It was giving an error otherwise as the environment variable is missing

Comment (Verse1) [reply front-end/src/pages/profile.js:109]: Once you create your .env file and add the variables you should rerun the dev script for it to recompile

File: front-end/src/styles/Navigation.css (hunk 2 above)
Comment (Verse1) [reply front-end/src/styles/Navigation.css:5]: Yes, the width and its positioning

File: front-end/src/components/SearchBar.jsx (hunk 3 above)
Comment (Verse1) [reply front-end/src/components/SearchBar.jsx:7]: For now, you can create the autocompletion of the search getting the classes from the api

File: front-end/src/pages/profile.js (hunk 1 above)
Comment (Star3Lord) [reply front-end/src/pages/profile.js:109]: the image url shouldn't be a environment variable. We should have a file containing all backend api urls as constants. The backend base url would be a environment variable.

Comment (Verse1) [reply front-end/src/pages/profile.js:109]: And that is exactly what we are doing, but we are outsourcing those images from an external API that is why it is just temporary

File: front-end/src/styles/Navigation.css (hunk 2 above)
Comment (Star3Lord) [reply front-end/src/styles/Navigation.css:5]: the width and positioning is controlled in the container

File: front-end/src/components/SearchBar.jsx (hunk 3 above)
Comment (Star3Lord) [reply front-end/src/components/SearchBar.jsx:7]: will update once the backend is setup

**Comment by Verse1** [comment]
You're a couple of commits behind the master branch

File: front-end/src/components/SearchBar.jsx (hunk 3 above)
Comment (Verse1) [reply front-end/src/components/SearchBar.jsx:7]: No problem

File: front-end/src/styles/Navigation.css (hunk 2 above)
Comment (Verse1) [reply front-end/src/styles/Navigation.css:5]: Regardless, it's positioning is too wide on the nav bar

**Comment by Star3Lord** [comment]
> You're a couple of commits behind the master branch

it's because you added api_url variable to .env

**Comment by Verse1** [comment]
> > You're a couple of commits behind the master branch
> 
> it's because you added api_url variable to .env

There were other commits as well

**Comment by Star3Lord** [comment]
> > > You're a couple of commits behind the master branch
> > 
> > 
> > it's because you added api_url variable to .env
> 
> There were other commits as well

yeah, so don't commit directly, make pull requests. revert those commit.

**Comment by Verse1** [comment]
> > > > You're a couple of commits behind the master branch
> > > 
> > > 
> > > it's because you added api_url variable to .env
> > 
> > 
> > There were other commits as well
> 
> yeah, so don't commit directly, make pull requests. revert those commit.

All commits were minor changes such as making the notifications panel smaller etc. So they are not new features in and of themselves. For those, new branches are created of course.

**Pull request merged**